test/nested-document-path.test.ts

```diff
@@ -19,13 +22,20 @@ describe('nested document path', function() {
     schema = await getSchema(docs);
   });
 
-  it('should assemble the path correctly with dot-notation', function() {
+  it('should assemble the path correctly fields with dots in their names', function() {
     const foo = schema.fields.find(v => v.name === 'foo');
     const bar = (foo?.types.find(v => v.name === 'Document') as DocumentSchemaType)?.fields.find(v => v.name === 'bar');
     const baz = (bar?.types.find(v => v.name === 'Document') as DocumentSchemaType)?.fields.find(v => v.name === 'baz');
     assert.ok(foo);
     assert.ok(bar);
     assert.ok(baz);
-    assert.equal(baz.path, 'foo.bar.baz');
+    assert.deepEqual(baz.path, ['foo', 'bar', 'baz']);
```

Comment:
Is this right? Elsewhere where paths are arrays the elements in the array are the object keys that make up the path to that thing.

ie. for this doc:

```javascript
{
  'field.with.path': {
    a: 1,
    b: 2
  }
}
```

The unique paths are:

```javascript
[
  ['field.with.path'],
  ['field.with.path', 'a'],
  ['field.with.path', 'b']
]
```

The reason it is usually an array of field names (at least in the new import/export) is specifically to disambiguate `{ 'field.with.path': 1 }` from `{ field: { with: { path: 1 }}}` because if those were strings with everything just joined together with dots both would have the path `'field.with.path'` 

Maybe I'm totally misreading what's going on here.

Reply:
I mean, that’s exactly what’s going on here, right? The doc is

```js

    {
      foo: {
        bar: {
          baz: 1
        },
        'bar.with.dot': {
          bah: 2
        }
      }
    }
```

so the paths are

```js
assert.deepEqual(baz.path, ['foo', 'bar', 'baz']);
assert.deepEqual(bah.path, ['foo', 'bar.with.dot', 'bah']);
```

Reply:
OK I'm confused, then. Carry on.